### ISSUE TITLE
Test  Android-CPP

### ISSUE THREAD

**Issue by wanghuan1115** [sdkbox/sdkbox-vungle-sample#6]
### 编译出错

错误信息:

~~~
-pre-compile:

-compile:
    [javac] Compiling 4 source files to /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/proj.android/bin/classes
    [javac] 警告: [options] 源值1.5已过时, 将在未来所有发行版中删除
    [javac] 警告: [options] 目标值1.5已过时, 将在未来所有发行版中删除
    [javac] 警告: [options] 要隐藏有关已过时选项的警告, 请使用 -Xlint:-options。
    [javac] 3 个警告

-post-compile:

-obfuscate:

-dex:
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/proj.android/bin/classes
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/bin/classes.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/google-play-services_lib/bin/classes.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/javax.inject-1.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/vungle-publisher-adaptive-id-3.3.2.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/PluginVungle.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/dagger-1.2.2.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/nineoldandroids-2.4.0.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/support-v4-18.0.0.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/google-play-services_lib/libs/google-play-services.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/sdkbox.jar
      [dex] input: /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/vungle-publisher-adaptive-id-3.3.3.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/bin/classes.jar -> classes-9d4d652451be3da5c992fdd3578873f0.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/google-play-services_lib/bin/classes.jar -> classes-72711d05ae7594cb93e04447d61928dc.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/javax.inject-1.jar -> javax.inject-1-325d44481e2894b25af05dc852fd7514.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/vungle-publisher-adaptive-id-3.3.2.jar -> vungle-publisher-adaptive-id-3.3.2-2d3668342aa7c4b60aba9676fef4ef01.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/PluginVungle.jar -> PluginVungle-9f710cf61817d806a30f6784da95d6df.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/dagger-1.2.2.jar -> dagger-1.2.2-355e50a950bc04167361f72acb3bcdf0.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/nineoldandroids-2.4.0.jar -> nineoldandroids-2.4.0-a2ef54bb8a916670d5233e3814b32f97.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/support-v4-18.0.0.jar -> support-v4-18.0.0-44109ccb413e1713548c3d6cdf246801.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/google-play-services_lib/libs/google-play-services.jar -> google-play-services-346852ef11e34d49b00103be36274cff.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/sdkbox.jar -> sdkbox-ffbc189ffbfdb5a9f9457f1eecb76b60.jar
      [dex] Pre-Dexing /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/cocos2d/cocos/platform/android/java/libs/vungle-publisher-adaptive-id-3.3.3.jar -> vungle-publisher-adaptive-id-3.3.3-ce61ccb2fdc4c2a14dc8722bb5f0fcb3.jar
      [dex] Converting compiled files and external libraries into /Users/gbox/Documents/chukong_test/sdkbox-vungle-sample/cpp/proj.android/bin/classes.dex...
       [dx] Merged dex A (22 defs/20.3KiB) with dex B (88 defs/111.1KiB). Result is 110 defs/154.3KiB. Took 0.1s
       [dx] Merged dex A (110 defs/154.3KiB) with dex B (1 defs/0.4KiB). Result is 111 defs/192.9KiB. Took 0.0s
       [dx] Merged dex A (111 defs/192.9KiB) with dex B (6 defs/1.6KiB). Result is 117 defs/261.1KiB. Took 0.0s
       [dx] Merged dex A (117 defs/261.1KiB) with dex B (714 defs/547.3KiB). Result is 831 defs/1044.6KiB. Took 0.1s
       [dx] Merged dex A (831 defs/1044.6KiB) with dex B (6 defs/6.3KiB). Result is 837 defs/1459.6KiB. Took 0.1s
       [dx] Result compacted from 2275.8KiB to 721.6KiB to save 1554.1KiB
       [dx] Merged dex A (837 defs/1459.6KiB) with dex B (44 defs/53.8KiB). Result is 881 defs/721.6KiB. Took 0.1s
       [dx] Merged dex A (881 defs/721.6KiB) with dex B (73 defs/100.8KiB). Result is 954 defs/1001.3KiB. Took 0.0s
       [dx] Merged dex A (954 defs/1001.3KiB) with dex B (444 defs/491.7KiB). Result is 1398 defs/1895.2KiB. Took 0.1s
       [dx] Result compacted from 8276.9KiB to 6107.8KiB to save 2169.1KiB
       [dx] Merged dex A (1398 defs/1895.2KiB) with dex B (5610 defs/4845.3KiB). Result is 7008 defs/6107.8KiB. Took 0.7s
       [dx] Result compacted from 7329.5KiB to 6144.1KiB to save 1185.4KiB
       [dx] Merged dex A (7008 defs/6107.8KiB) with dex B (38 defs/47.0KiB). Result is 7046 defs/6144.1KiB. Took 0.4s
       [dx]
       [dx] UNEXPECTED TOP-LEVEL EXCEPTION:
       [dx] com.android.dex.DexException: Multiple dex files define Lcom/vungle/log/Logger;
       [dx] 	at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:594)
       [dx] 	at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:552)
       [dx] 	at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:533)
       [dx] 	at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:170)
       [dx] 	at com.android.dx.merge.DexMerger.merge(DexMerger.java:188)
       [dx] 	at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:439)
       [dx] 	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:287)
       [dx] 	at com.android.dx.command.dexer.Main.run(Main.java:230)
       [dx] 	at com.android.dx.command.dexer.Main.main(Main.java:199)
       [dx] 	at com.android.dx.command.Main.main(Main.java:103)
       [dx]

BUILD FAILED
/Users/gbox/Android/sdk/tools/ant/build.xml:888: The following error occurred while executing this line:
/Users/gbox/Android/sdk/tools/ant/build.xml:890: The following error occurred while executing this line:
/Users/gbox/Android/sdk/tools/ant/build.xml:902: The following error occurred while executing this line:
/Users/gbox/Android/sdk/tools/ant/build.xml:283: null returned: 2

Total time: 54 seconds
执行命令出错，返回值：1。
~~~

**Comment by yinjimmy** [comment]
More information, please.

**Comment by yinjimmy** [comment]
fixed with commit 60fe3bbd647097b843b83f55f9dcadb87aef27ee
